### DISCUSSION TITLE
Looking for help with redirect please

### DISCUSSION THREAD

**Discussion by darrylmorley** [caprover/caprover#1596] [Q&A]
Hi,

I'm trying to forward all traffic from our non www domain to the www domain. I have the below rule set in Caprover, but it doesn't appear to be working correctly.

Typing mydomain.co.uk into the address bar in the browser will forward correctly to https://www.mydomain.co.uk/.

But, visiting https://mydomain.co.uk/ does not forward and gives a security warning.

```
server {
    listen 443 ssl;
    server_name mydomain.co.uk;
    ssl_certificate     /captain/data/letencrypt/etc/live/www.mydomain.co.uk/fullchain.pem;
    ssl_certificate_key /captain/data/letencrypt/etc/live/www.mydomain.co.uk/privkey.pem;   
    return 301 https://www.mydomain.co.uk$request_uri;
}

```

How do I get https://mydomain.co.uk/ to forward to https://www.mydomain.co.uk/?

**Comment by maietta** [comment]
Option 1) RECOMMENDED: Remove the non-www from your app and create a new Nginx redirect app and attach that domain. Enable SSL for that and redirect to your www version.

Option 2) Enable domain and SSL on your main app, then in your application's code, detect the URL being loaded and redirect accordingly. You'll need to add a Service Override to pass the host information to the application. I don't have that handy and I don't recommend this anyway.

Option 3) If your domain is hosted on CloudFlare or another front-end caching proxy provider, you can configure SSL and redirects via their system's dashboard.

**Reply by darrylmorley** [reply]
You shouldn't really have to do that though should you, you should be able to configure this from the nginx conf surely?

**Reply by maietta** [reply]
> You shouldn't really have to do that though should you, you should be able to configure this from the nginx conf surely?

Option 4) If you are comfortable with the risk of breaking your config, modifying your nginx config is certainly an option, but SSL won't be automatic here.

**Reply by githubsaturn** [reply]
> You shouldn't really have to do that though should you, you should be able to configure this from the nginx conf surely?

Technically yes, but if you're not an expert with nginx, then it's the best option. Also keep in mind that other than a new row in your app list it really does not use any memory resources (perhaps 1-2mb of RAM)!
